### PR TITLE
Tests for middlewares/auth

### PR DESCRIPTION
**Pull request by pieterlukasse**
Jira Ticket: [PXP-9844](https://ctds-planx.atlassian.net/browse/PXP-9844)

### Improvements
- reorganized `middlewares/auth.go` code and added some tests for it
- updated README with info on deployment to QA
- updated README with info about updating Docker dependencies

### CODE REVIEW

File: tests/middlewares_tests/middlewares_test.go [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+func TestPrepareNewArboristRequest(t *testing.T) {
+	setUp(t)
+	requestContext := new(gin.Context)
+	requestContext.Params = append(requestContext.Params, gin.Param{Key: "Authorization", Value: "dummy_token_value"})
+	requestContext.Writer = new(tests.CustomResponseWriter)
+	requestContext.Request = new(http.Request)
+	requestContext.Request.Header = map[string][]string{
+		"Authorization": {"dummy_token_value"},
+	}
+	u, _ := url.Parse("https://cohort-middleware/api/abc/123")
```
Comment (pieterlukasse) [inline tests/middlewares_tests/middlewares_test.go:52]: should this be a relative path? Maybe yes...otherwise line 57 below is a bit strange...

Comment (m0nhawk) [reply tests/middlewares_tests/middlewares_test.go:52]: Yes, it should be relative, so in this case it'll use `/api/abc/123` for authentication, and resource will have prefix `/cohort-middleware` as an extra part.

Though, for deploying behind reverse proxy, it should use the same part, ignoring all the subpaths under which cohort-middleware is exposed, and, it currently works like that in [cloud-automation](https://github.com/uc-cdis/cloud-automation/blob/890867834f33a87cb58772f7203d9c8b95165b19/kube/services/revproxy/gen3.nginx.conf/cohort-middleware-service.conf#L5).

**Review by m0nhawk — APPROVED** [review]
lgtm 👍

File: tests/middlewares_tests/middlewares_test.go (hunk 1 above)
Comment (pieterlukasse) [reply tests/middlewares_tests/middlewares_test.go:52]: Ok, the test seems correct to me. Thanks for the info.

**Pull request merged**